common/hyperscaler/gardener_client.go

```diff
@@ -0,0 +1,112 @@
+package hyperscaler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-project/kyma-environment-broker/common/gardener"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+func (sp *GardenerClient) IsUsedByShoot(secretBinding *gardener.SecretBinding) (bool, error) {
```

Comment:
I know this is legacy behavior but for every `SecretBinding` we list all shoots.